libflux/src/core/semantic/builtins.rs

```diff
@@ -49,7 +49,7 @@ pub fn builtins() -> Builtins<'static> {
                  "millisecond" => "forall [] (t: time) -> int",
                  "microsecond" => "forall [] (t: time) -> int",
                  "nanosecond" => "forall [] (t: time) -> int",
-                 "truncate" => "forall [] (t: time, unit: duration) -> time",
+                 "truncate" => "forall [t0] where t0 : Timeable (t: t0, unit: duration) -> time", // sahas
```

Comment:
Can you remove the terminating comment?

Reply:
Also is there a reason you updated just truncate? I thought the type signatures would be updated all at once along with the implementations.

Reply:
@jlapacik I was not sure if I would be able to implement the Kind constraint so I just tried one function to see if it would. Next step of mine is to apply the change to all functions where it is applied.

About the comment, I will remove it. I wanted a flag to search for everywhere I made a change to so I can go back and revert if necessary

Reply:
@ChitlangeSahas I'd like to see this work done in at least two parts. 1. A change that adds timeable as a new constraints with test etc. 2. A change that adds it to date.truncate and possible the other date functions. I'd even be ok with each date function getting its own PR so we can review smaller changes at a time.

So for this PR I would focus on getting the unit tests in place for the Timeable constraint. 

Does that make sense?

Reply:
Yeah for sure, I will change the signature back to what it was.